sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -742,12 +750,40 @@ object ScalaJSPluginInternal {
 
   // These settings will be filtered by the stage dummy tasks
   val scalaJSRunSettings = Seq(
+      scalaJSMainEntryPoint := {
+        if (scalaJSUseMainEntryPoint.value) {
+          mainClass.value.fold {
+            throw new MessageOnlyException(
+                "No main class class detected, but scalaJSUseMainEntryPoint " +
+                "was set to true. If you do not want to derive a main entry " +
+                "point from discovered main classes, add " +
+                "`scalaJSMainEntryPoint := None`")
+          } { mainClass =>
+            Some(EntryPoint.mainMethod(mainClass, "main"))
+          }
+        } else {
+          None
+        }
+      },
```

Comment:
So in fact I kept it, because it can be useful if the developer has a non-`JSApp` module initializer. They want to `run` their app (so `scalaJSUseMainModuleInitializer` is true) but they do not want to have an automatically derived module initializer (so `scalaJSMainModuleInitializer` is customized).

Moreover, we don't have an assumption `scalaJSUseMainModuleInitializer == scalaJSMainModuleInitializer.isDefined`. It's perfectly OK to have `None` when `scalaJSUseMainModuleInitializer` is true.